fairscale/nn/misc/flatten_params_wrapper.py

```diff
@@ -110,18 +133,35 @@ def _unflatten_params(self) -> None:
         del self.flat_param
 
     def _unflatten_params_as_views(self) -> None:
+        assert self.is_flattened
         ps = self._get_param_views()
         for (m, n), p in zip(self._param_infos, ps):
             setattr(m, n, p)  # This will set as plain attr
         for (m, n, shared_m, shared_n) in self._shared_param_infos:
             setattr(m, n, getattr(shared_m, shared_n))
 
     @contextmanager
-    def unflatten_params(self) -> Generator:
-        self._unflatten_params()
+    def unflatten_params(self, recurse: bool = True) -> Generator:
+        """
+        Unflatten params (optionally recursively on all nested instances).
+        If the current instance is already unflattened, then it will remain
+        unflattened after the context manager exits.
+        """
+        if recurse:
+            with ExitStack() as stack:
+                for module in self.modules():
+                    if isinstance(module, FlattenParamsWrapper):
+                        stack.enter_context(module.unflatten_params(recurse=False))
+                yield
+            return
```

Comment:
how about change this to a "else:"? early return can cause future common code to be skipped by mistaken.